rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/OneScanActivity.java

```diff
@@ -134,7 +189,9 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         oneScanHelper.doActivityResult(requestCode, resultCode, data);
         if (resultCode == Activity.RESULT_CANCELED) {
-            onBackPressed();
+            if (!enableBatchScan || dataArray.length() <= 0) {
```

Comment:
You can use `Utils.isEmptyJsonArray` in place of `dataArray.length() <= 0`

Reply:
fixed.